src/celeritas/global/CoreState.cc

```diff
@@ -18,6 +18,10 @@
 
 namespace celeritas
 {
+//---------------------------------------------------------------------------//
+//! Support polymorphic deletion
+CoreStateInterface::~CoreStateInterface() = default;
```

Comment:
I've been doing this as a matter of habit for pure abstract classes for some time to avoid "weak vtables": 
In [GCC](https://gcc.gnu.org/onlinedocs/gcc/Vague-Linkage.html):
>  If the class declares any non-inline, non-pure virtual functions, the first one is chosen as the “key method” for the class, and the vtable is only emitted in the translation unit where the key method is defined.

I think it's less important if the hierarchy is only used within a single library, and more important in a case like this where a potentially arbitrary number of libraries subclasses an interface.

See https://stackoverflow.com/questions/12024642/placing-of-external-virtual-tables/12025816#12025816 , https://stackoverflow.com/a/23749273/118160 .